cmd/osbuild-jobsite-manager/main.go

```diff
@@ -0,0 +1,387 @@
+				errs <- err
+				return
+			}
+
+			defer res.Body.Close()
+
+			if res.StatusCode != http.StatusOK {
+				errs <- fmt.Errorf("StepClaim: Got an unexpected response %d while expecting %d. Exiting.", res.StatusCode, http.StatusOK)
+				return
+			}
+
+			break
+		}
+
+		logrus.Info("StepClaim: Done.")
+
+		close(done)
+	})
+}
+
+func StepProvision(manifest []byte) error {
+	return Wait(argTimeoutProvision, func(done chan<- struct{}, errs chan<- error) {
+		res, err := Request("PUT", "provision", manifest)
+
+		if err != nil {
+			errs <- err
+			return
+		}
+
+		defer res.Body.Close()
+
+		if res.StatusCode != http.StatusCreated {
+			errs <- fmt.Errorf("StepProvision: Got an unexpected response %d while expecting %d. Exiting.", res.StatusCode, http.StatusCreated)
+			return
+		}
+
+		logrus.Info("StepProvision: Done.")
+
+		close(done)
+	})
+}
+
+func StepPopulate() error {
+	return Wait(argTimeoutPopulate, func(done chan<- struct{}, errs chan<- error) {
+		res, err := Request("POST", "populate", []byte(""))
+
+		if err != nil {
+			errs <- err
+			return
+		}
+
+		defer res.Body.Close()
+
+		if res.StatusCode != http.StatusOK {
+			errs <- fmt.Errorf("StepPopulate: Got an unexpected response %d while expecting %d. Exiting.", res.StatusCode, http.StatusOK)
+			return
+		}
+
+		logrus.Info("StepPopulate: Done.")
+
+		close(done)
+	})
+}
+
+func StepBuild() error {
+	return Wait(argTimeoutBuild, func(done chan<- struct{}, errs chan<- error) {
+		arg := BuildRequest{
+			Pipelines:    argPipelines,
+			Environments: argEnvironments,
+		}
+
+		dat, err := json.Marshal(arg)
+
+		if err != nil {
+			logrus.Fatalf("StepBuild: Failed to marshal data: %s", err)
+		}
+
+		res, err := Request("POST", "build", dat)
+
+		if err != nil {
+			errs <- err
+			return
+		}
+
+		defer res.Body.Close()
+
+		if res.StatusCode != http.StatusCreated {
+			errs <- fmt.Errorf("StepBuild: Got an unexpected response %d while expecting %d. Exiting.", res.StatusCode, http.StatusOK)
+			return
+		}
+
+		logrus.Info("StepBuild: Done.")
+
+		close(done)
+	})
+}
+
+func StepProgress() error {
+	return Wait(argTimeoutProgress, func(done chan<- struct{}, errs chan<- error) {
+		for {
+			res, err := Request("GET", "progress", []byte(""))
+
+			if err != nil {
+				errs <- err
+				return
+			}
+
+			defer res.Body.Close()
+
+			if res.StatusCode == http.StatusAccepted {
+				logrus.Info("StepProgress: Build is pending. Retry.")
+				time.Sleep(5 * time.Second)
+				continue
+			}
+
+			if res.StatusCode != http.StatusOK {
+				errs <- fmt.Errorf("StepProgress: Got an unexpected response %d while expecting %d. Exiting.", res.StatusCode, http.StatusOK)
+				return
+			}
+
+			break
+		}
+
+		logrus.Info("StepProgress: Done.")
+
+		close(done)
+	})
+}
+
+func StepExport() error {
+	return Wait(argTimeoutExport, func(done chan<- struct{}, errs chan<- error) {
+		for _, export := range argExports {
+			res, err := Request("GET", fmt.Sprintf("export?path=%s", export), []byte(""))
```

Comment:
This needs [`PathEscape`](https://pkg.go.dev/net/url#PathEscape)

Reply:
Done!